File: arrays/containsduplicate/py/containsduplicate.py
from typing import List

# https://leetcode.com/problems/contains-duplicate
class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:

        hashset = set()

        for n in nums:
            if n in hashset:
                return True

            hashset.add(n)

        return False

obj = Solution()

nums = [1, 2, 3]
r = obj.containsDuplicate(nums)
print(r)

File: arrays/containsduplicate/go/containsduplicate.go
package main

import "fmt"

func main() {
	nums := []int{1, 2, 3, 1}
	res := containsDuplicate(nums)
	//should print true. since at least one element appears twice.
	fmt.Println(res)
}

func containsDuplicate(nums []int) bool {

	hashset := make(map[int]bool)

	for _, elm := range nums {
		if hashset[elm] {
			return true
		}
		hashset[elm] = true
	}
	return false
}
